weave/wandb_interface/wandb_stream_table.py

```diff
@@ -0,0 +1,249 @@
+import contextlib
+import json
+import logging
+import os
+import random
+import shutil
+import typing
+import uuid
+
+from wandb.sdk.lib.paths import LogicalPath
+
+from .wandb_lite_run import InMemoryLazyLiteRun
+
+from .. import runfiles_wandb
+from .. import storage
+from .. import weave_types
+from .. import artifact_base
+from .. import file_util
+
+if typing.TYPE_CHECKING:
+    from wandb.sdk.internal.file_pusher import FilePusher
+
+# Shawn recommended we only encode leafs, but in my testing, nested structures
+# are not handled as well in in gorilla and we can do better using just weave.
+# Uncomment the below to use gorilla for nested structures.
+TRUST_GORILLA_FOR_NESTED_STRUCTURES = False
```

Comment:
If gorilla doesn't handle this, there will be a perf issue for nested data, because Gorilla is in charge of compaction to parquet.

Reply:
the primary issue i noticed that the history keys returned for a nested structure walk as far down as they can. For example, if you log `{'a':{'b': 3}}`, gorilla treats it as a single column at `a.b`. I can look into this. It is possible that we just need to interpret the gorilla history types a bit more carefully and reverse construct our top level keys.